### ISSUE TITLE
Switching credential during provisioning for Windows servers

### ISSUE THREAD

**Issue by wilber82** [hashicorp/packer#9664]
My image configuration process has 3 steps:

1. Basic configure OS via PowerShell script by local 'Administrator' account. (For example, set timezone)
2. Join the VM to domain by PowerShell script.
3. Install a software via PowerShell script by **domain account**.
4. Leave domain via PowerShell script.
5. Continue configure OS via PowerShell by local 'Administrator' account.

As you can see. It needs domain account to connect the server.
Looks like there is no way to switching credential temporarily during provisioning.

I tried to give "elevated_user" and "elevated_password" in PowerShell block. No lucky.

Even I don't give any additional keys, the audit log shows Packer tries to login the machine by 'Administrator@domainName.com" but not local 'administrator' account. It always failed as I'm not domain administrator.

Is this a bug, or?

My builders block is correct. I'm using winrm as communicator. it works fine for any VM without the need to switch account.

**Comment by SwampDragons** [comment]
You're right. Currently there is no way to change the communicator credentials halfway through a build.

**Comment by kevincarr24** [comment]
@allaboutopensource How can you run a script during the build process? I have seen that custom_script is available, but I have failed to get this to run properly on my build images. I understand that this is an older comment but any advice is greatly appreciated.
